File: 3099-harshad-number/3099-harshad-number.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int orig=x;
        int sum=0;
        while(x>0){
            int r=x%10;
            sum+=r;
            x=x/10;
        }
        if(orig%sum==0){
            return sum;
        }
        else{
            return -1;
        }
        
    }
}

File: 3099-harshad-number/3099-harshad-number.py
class Solution(object):
    def sumOfTheDigitsOfHarshadNumber(self, x):
        sum=0
        original=x
        while x>0:
            r=x%10
            sum+=r
            x=x//10
        if original%sum==0:
            return sum
        else:
            return -1
        